### DISCUSSION TITLE
EAGE-HACK-2024: Possible Hackathon Ideas

### DISCUSSION THREAD

**Discussion by aruss175** [EAGE-Annual-Hackathon/EAGE-Annual-Hackathon-Discussion-Boards#11] [Ideas]
Have ideas - post them here?  The organizers will also share some ideas and possible topics to hack on!

**Comment by aruss175** [comment]
Please find a gathering of ideas from the Hackathon conveners here:

**Big Data Workflows - Seismic and Simulation:**

- Convert a PINN / FNO propagator to Modulus and run multi-node in complex environment
- Run your own method at scale
- Solve inverse problem with physics-ML
- Enhance a solver
- Optimize seismic wave propogation or do it
with PINN / FNO

We will have tools from Nvidia to help with some of these ideas

**New Energy:**

- ResSim for CO2 modeling
- PINN for turbulent flow modeling
- Optimization of wind turbine
- positioning
- Predicting in data-poor areas for carbon capture and storage
- Improve algorithms for climate models

**Optimization and Reducing code footprint**

- Convert your code to 3D and show performance
- Optimization of existing open source packages or code (OSDU anyone?)

**Comment by ovcharenkoo** [comment]
**LLM for optimization of geoscience workflows**
* LoRa Fine-tuning of LLM on research papers and tools instructions
* Building evaluation datasets for niche domains (seismic, reservoir, chemical)
* Creating assistants with domain-tuned LLM for automation
* Embeddings for seismic data
* Understanding text and images for multimodality

**Reply by marcosjacinto** [reply]
Fine-tuning LLMs seems like an excellent idea and a good challenge

**Reply by aruss175** [reply]
Also interesting - new insider information on the carbon emissions of LLMs: https://towardsdatascience.com/the-carbon-footprint-of-gpt-4-d6c676eb21ae

**Reply by pederka** [reply]
This. If we can combine a PEFT-method with a suitably large domain dataset this would definitely be a feasible task. Any ideas on datasets?

**Reply by aruss175** [reply]
We have all the EAGE abstracts from last year...would that work?

**Comment by aruss175** [comment]
For anyone wanting to go the open source route - there are a set of python packages here that would LOVE some help in optimization and speed!!

https://softwareunderground.org/stack

**Comment by roderickperez** [comment]
Hello everyone, is the Hackathon project already selected? I'd like to use **NVIDIA Modulus Simulationin a CCUS case** (something like was mentioned before in "ResSim for CO2 modeling")

**Reply by aruss175** [reply]
We expect that there will be a number of teams hacking on different projects - just have to show a contribution to netzero or energy reduction.  So if you can formulate your idea then you can "sell" it to get some people to work with you.  We will have time on Sunday morning to get the teams together.
